### ISSUE TITLE
the preference for a default application that is supposedly set in the 'Open with …' dialogue appears to be not persistent

### ISSUE THREAD

**Issue by grahamperrin** [trueos/lumina#373]
https://github.com/trueos/lumina/issues/362 was for the 'Open with …' feature of Lumina Screenshot. 

https://github.com/trueos/lumina/issues/372 observes an issue, unexpected appearance of that feature. 

When issue 372 occurs: 

1. select an application
2. check the box, [*] Set this application as the default
3. click 'OK'

## Expected

- a default; a preference that will persist. 

## Actual result

- subsequent use of Quick Save seems to not respect that preference; there is, again, a prompt to choose an application to open what was saved. 

It's hard to tell whether this is a bug. 

Was Quick Save designed to Open? If so, then the open must be quick, not slowed by repeated requirements to choose applications.

**Comment by q5sys** [comment]
Open with was never designed to be a permanent setting, and in modern operating systems "Open with..." always means that the user will be presented with a choice of what application to use.  Hence the difference between "Open" and "Open with..."  
Similar to how Lumina-Open will always present a list of options to start open a file.  The ideology behind that was a user may want to change what program he uses to open a file.  They may want to open GIMP to edit, or they may want to open a viewer, or they may want to open an email client  and send the file as an attachment.  The Open with dialog allows the user to select what they want at the time they want to open the file.   For some back story, originally I wrote it so that it'd always open with the same program, and Ken wanted to leave the option open to the user.

Ken wants to re-write Lumina-Open... so if there is further functionality that you want for opening options, it might be able to be added in then; but that's not an issue for this ticket.  Start a thread in the forum to see what all the community might be interested in, regarding Lumina-Open.

I addressed the issue with the save button issue in its own ticket.  I'll leave this open until I see Ken in a few days so he can comment if he wants to correct me on the intended implementation of "Open With..."

**Comment by utrenkner** [comment]
@q5sys several times already, I ran into exactly the same problem: I wanted to make a certain file type to always open in a certain application and thus checked the ``Set this application as the default``. But without any effect - the next time, I opened the same file type, some other app was used...

There clearly should be an easy way to set the default app for a file type. The "Open with..." dialog seemed to offer this, but it currently does not work as expected.

**Comment by beanpole135** [comment]
I just forgot to remove the "Quick Save" action when I converted the app to use the "Open With" button instead. I corrected this mistake.

https://github.com/trueos/lumina/commit/2d8f2ac508af99e6b8196c8dada3574ffdea3767

When lumina-open is run with the "-select" flag ("Open With" as opposed to "Open" functionality), it is supposed to treat the currently default app as the pre-selected option in the "Preferred" tab. I will go through and make sure that is still functioning properly here in a few minutes though.
